language-plutus-core/src/Language/PlutusCore/Type/Core.hs

```diff
@@ -0,0 +1,195 @@
+{-# LANGUAGE UndecidableInstances #-}
+{-# LANGUAGE DeriveAnyClass        #-}
+{-# LANGUAGE DerivingVia           #-}
+{-# LANGUAGE FlexibleInstances     #-}
+{-# LANGUAGE LambdaCase            #-}
+{-# LANGUAGE MultiParamTypeClasses #-}
+{-# LANGUAGE TypeFamilies          #-}
+
+module Language.PlutusCore.Type.Core
```

Comment:
Just look how nicely this module reads finally. Only relevant data types sorted in "chronological" order. Also a tiny bit of junk, but we can fix that later.